packages/types/definitions.ts

```diff
@@ -284,71 +291,7 @@ export type Scenario = {
    * @default 1
    */
   weight?: number;
-} & (
-  | {
-      /**
-       * @title HTTP engine
-       */
-      engine?: 'http';
-      /**
-       * @title Scenario flow
-       */
-      flow: Array<
```

Comment:
I think an optimal strategy may be to keep these supported properties typed and available on the scenario object but mark them all as optional. No property discrimination based on the engine but a typed experience for those who decide to use those properties.